pkg/kubelet/kubelet.go

```diff
@@ -3671,6 +3671,14 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *api.Pod, podStatus *kubeco
 
 	// Handle the containers failed to be started, which should be in Waiting state.
 	for _, container := range containers {
+		if hasInitContainers {
```

Comment:
isInitContainer is for when we are iterating over the list of init containers (so rules that apply to init containers apply when that boolean is true).  This method is called for both init containers and regular containers.